core/src/main/java/org/elasticsearch/http/netty/NettyHttpServerTransport.java

```diff
@@ -498,7 +509,10 @@ protected void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) thro
                 // ignore
                 return;
             }
-            if (!NetworkExceptionHelper.isCloseConnectionException(e.getCause())) {
+            if (e.getCause() instanceof CircuitBreakingException) {
```

Comment:
We need the special handling because `CircuitBreakingException` is thrown when the amount in-flight bytes exceeds a limit. [RFC 2616, section 10](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html) specifically states that this can be also used to signal temporary condition and that the server should provide a hint in that case (which we could do). While I agree with you that this may not be 100% correct I think this is the best we can do and clients would then be even able to base their retry logic on this status code which in line with the semantics of RFC 2616. What status code would you suggest in that case?